crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1105,6 +1111,71 @@ fn find_only_token_in_range(
     token
 }
 
+/// Attach an enclosed end-of-line comment to a [`StmtImportFrom`].
+///
+/// For example, given:
+/// ```python
+/// from foo import (  # comment
+///    bar,
+/// )
+/// ```
+///
+/// The comment will be attached to the `StmtImportFrom` node as a dangling comment, to ensure
+/// that it remains on the same line as the `StmtImportFrom` itself.
+fn handle_enclosed_import_from_comment<'a>(
+    comment: DecoratedComment<'a>,
+    import_from: &'a StmtImportFrom,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    let node = comment.enclosing_node();
+
+    // The comment needs to be on the same line, but before the first member, to triggering on:
+    // ```python
+    // from foo import (bar, baz,  # comment
+    //     qux,
+    // )
+    // ```
+    if comment.line_position().is_end_of_line()
+        && !locator.contains_line_break(TextRange::new(import_from.start(), comment.start()))
```

Comment:
Does this work with line continuation? 

```python
from a import \
	( # comment
		b,
		c
	)
```

It could make sense to search for any opening `(` parentheses instead and assert that the comment starts after the opening but before the start member. This may even be faster because the tokenizer only needs to eat over a few whitespace tokens.

Reply:
Good suggestion, changed.